### ISSUE TITLE
ERROR - diamond execution failed! diamond-error-code=-4

### ISSUE THREAD

**Issue by Clabe1986** [oschwengers/platon#10]
Hello,
After successfully installing platon with conda as remote user on the server, I was very excited of running my samples. However, the above error has persisted and I seem not to head anywhere in fixing it. 
Here is the script:
platon --db db --output platon_S2 --verbose S2.fasta

And here is the error
![output](https://user-images.githubusercontent.com/22397564/87218322-e2058000-c351-11ea-81b2-2a2b3098e811.png)
 Is there anyway I can fix this? 
I am using contigs assembled with spades

**Comment by oschwengers** [comment]
Hi @Clabe1986 , thanks for reporting this issue. Unfortunately, so far I cannot reproduce it. Have you downloaded the latest version of the database? `wget https://zenodo.org/record/3924529/files/db.tar.gz`
The previous version contained some files with wrong file permissions. Maybe this is somehow related.
If this is not the case, could you please provide some more information, _i.e._ the entire `S2.log` file, the platon version and (if possible) the query genome, so that I can try to reproduce the error?

**Comment by oschwengers** [comment]
Hi @Clabe1986 , in addition, could you also please execute the exact `diamond` version within the `bioconda` environment on the same machine either some test data or the intermediate proteins? Maybe this bug is somehow related to the compiled `diamond` version instead of `platon` itself.

**Comment by Clabe1986** [comment]
Dear Oschwengers,
Sorry I have taken long to get back. I was trying to investigate the issues you had raised. To begin with, I am running python 3.7.6 on conda 4.8.3. I have successifully installed platon 1.4. 
I think diamond is the source of my problems, I have just tried to run diamond alone as you had advised and the output was:
$(base) [ro59qey@meggy B3_results]$ diamond --version
Illegal instruction
$(base) [ro59qey@meggy B3_results]$ which diamond
~/anaconda3/bin/diamond
&(base) [ro59qey@meggy B3_results]$ diamond --help
Illegal instruction

If I try installing diamond with this script:
$conda install -c bioconda/label/cf201901 diamond
instead of installing lates diamond 0.9.36, it instead installs diamond 0.9.24. 

I also tried installing diamond from the source:
$wget http://github.com/bbuchfink/diamond/releases/download/v0.9.36/diamond-linux64.tar.gz
     then,
$tar xzf diamond-linux64.tar.gz
On running it, 
$(base) [ro59qey@meggy ~]$ ./diamond --version
./diamond: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by ./diamond)

I have also installed platon with conda on my personal computer and there diamond works well, only that because of its low memory, I cannot run platon on it. I am currently using a university server and it seems it does not support latest versions of diamond. I have written to the admin about it and ones they have ensured that I run diamond 0.9.32, I will report any further progress or advice.
Thank you,
Clabe Wekesa

**Comment by oschwengers** [comment]
Hello @Clabe1986 , thanks for this comprehensive report. This indeed looks like a `diamond` specific hardware/os issue on your IT system. As you've been able to execute `platon` without issues on your personal computer I'd like to close this issue as it seems to be unrelated to `platon` itself.

However, please feel free to keep posting updates to your issue here or to reopen it if this might be of interest for other users.
Best regards and thanks again for your report.

**Comment by oschwengers** [comment]
@Clabe1986 
FYI, there's another issue on failed `diamond` executions: #12
I'll provide a fix for this upon successful tests.

**Comment by oschwengers** [comment]
@Clabe1986  Indeed, there seemed to be an issue with `diamond`. I successfully tested `platon` with the recent `diamond` version `2.0.4` on Linux. An update is pushed to BioConda and already public. Could you maybe give it a try and test whether this fixes the issue you've faced?

**Comment by mdtorohernando** [comment]
Hi Oliver! I'm working with platon (v1.7) and I have the same problem with diamond (diamond version 2.1.9) on an Ubuntu 20.04.3 LTS. 
I'm not sure what's happening but it happens with all the genomes I've tested: 
```

2024-06-03 18:10:06,200 - INFO - CONFIG - mode=accuracy
2024-06-03 18:10:06,200 - INFO - CONFIG - characterize=False
2024-06-03 18:10:06,200 - INFO - CONFIG - metagenome=False
2024-06-03 18:10:06,209 - INFO - UTILS - dependency check: tool=prodigal, version=v2.6.3
2024-06-03 18:10:06,215 - INFO - UTILS - dependency check: tool=diamond, version=v2.1.9
2024-06-03 18:10:06,318 - INFO - UTILS - dependency check: tool=blastn, version=v2.15.0
2024-06-03 18:10:06,320 - INFO - UTILS - dependency check: tool=hmmsearch, version=v3.4.0
2024-06-03 18:10:06,322 - INFO - UTILS - dependency check: tool=nucmer, version=v4.0.0
2024-06-03 18:10:06,328 - INFO - UTILS - dependency check: tool=cmscan, version=v1.1.5
2024-06-03 18:10:06,347 - INFO - MAIN - exclude contig: too long: id=contig_1, length=3282054
2024-06-03 18:10:06,349 - INFO - MAIN - length contig filter: # input=4, # discarded=1, # remaining=3
2024-06-03 18:10:12,218 - INFO - MAIN - ORF detection: # ORFs=68
2024-06-03 18:10:12,218 - INFO - MAIN - ORF contig filter disabled! # passed contigs=3
2024-06-03 18:10:20,128 - ERROR - MAIN - diamond execution failed! diamond-error-code=-11


```

thanks in advance!

**Comment by mdtorohernando** [comment]
solved! The latest version of platon v1.7 fails with the latest version of diamond v2.1.9... so... downgrading in conda into the diamond v2.1.8 it works properly!!
